src/state/machines/selectors.ts

```diff
@@ -75,4 +75,17 @@ export const MachineSelectors = {
   },
 
   exhaustedCards: ({ context: { turn } }: { context: GameState }) => turn.exhaustedCards,
+
+  numberOfAnimalsBought: ({ context: { players } }: { context: GameState }) =>
+    players.flatMap((player) =>
+      [player.hand, player.deck, player.discard].flatMap((deck) => deck.filter((card) => card.type === "animal")),
+    ).length,
```

Comment:
I wonder can we easily "compose" selectors. To select things like `allPlayerCards` and build numberOfAnimalsBought/numberOfPlantsBought on top of that

Reply:
While "composing" is not really possible in the component that uses the selector, I implemented a halway solution now where we re-use an allPlayerCards selector in the number of animals/plants bought selectors